### DISCUSSION TITLE
When and how is the $logIndex variable in class LogReader stored?

### DISCUSSION THREAD

**Discussion by truongbo17** [opcodesio/log-viewer#69] [Q&A] [answered]
I can't find the code when saving the value to know what $logIndex looks like . I really don't know how you got the index of each line of its logs.

This is when i dd($logIndex) : 
![Screenshot from 2022-09-03 15-26-36](https://user-images.githubusercontent.com/91400591/188262761-c944acda-13f8-4a9c-be71-0df92e39c1dd.png)
![Screenshot from 2022-09-03 15-27-43](https://user-images.githubusercontent.com/91400591/188262776-35022da5-489e-4081-a17b-3c8bcaaacd21.png)

**Comment by arukompas** [comment, marked as answer]
Hey @truongbo17 , 

Check the `LogReader::close()` method - https://github.com/opcodesio/log-viewer/blob/main/src/LogReader.php#L215

When the file is closed (this means we're finished with it and we're cleaning up), it checks whether the index has changed. If so - it writes it into cache. It uses Laravel's `Cache` facade, so it uses your app's default cache driver (file, redis, etc)

When opening the file for the second time, the index is loaded from the cache (see the `LogReader::open()` method). If the file hasn't changed, then the index does not change either and upon closing it does not write into cache again.

Hope this makes sense! :)

*P.S. There are some changes coming later to how the index is built up, so please don't make any PRs related to indexing for now.*

**Reply by truongbo17** [reply]
Thanks for your detailed answer. I have read through your code. My real problem is why are you getting the index of those lines of logs.
What algorithm or code did you use to get the exact index position of the logs as shown below ?

![image](https://user-images.githubusercontent.com/91400591/188270325-d1db3ea3-b31c-4a97-8b34-54fe82781819.png)

**Reply by arukompas** [reply]
Check the `LogReader::scan()` method. 

It goes through the whole file and indexes the position whenever it finds a new log entry.

**Reply by truongbo17** [reply]
thanks , i will try it
